test/show.jl

```diff
@@ -808,6 +808,13 @@ end
 let repr = sprint(dump, Union{Integer, Float32})
     @test repr == "Union{Integer, Float32}\n" || repr == "Union{Float32, Integer}\n"
 end
+module M30442
+struct T end
+end
+let repr = sprint(show, Union{String, M30442.T})
+    @test (startswith(repr, "Union{")         && endswith(repr, "M30442.T, String}")) ||
```

Comment:
Do we really need to test both orders? It's not supposed to change AFAIK. Also you should be able to use `$(curmod_prefix)` to avoid the need for `endswith` (see examples elsewhere in the file).

Reply:
Afaik, display order isn't strictly defined, so I included both cases, following the style from L808-180. Just updated tests to use `curmod_prefix`, though.